pymemcache/client/base.py

```diff
@@ -96,11 +95,11 @@ def _parse_hex(value):
 def check_key_helper(key, allow_unicode_keys, key_prefix=b''):
     """Checks key and add key_prefix."""
     if allow_unicode_keys:
-        if isinstance(key, six.text_type):
+        if isinstance(key, str):
             key = key.encode('utf8')
-    elif isinstance(key, six.string_types):
+    elif isinstance(key, str):
         try:
-            if isinstance(key, six.binary_type):
+            if isinstance(key, bytes):
                 key = key.decode().encode('ascii')
```

Comment:
Few days ago I added tests against this function, they allow us to run fine grained tests against this function and they allow us to see what is covered. By looking at our code coverage we can observe that we never pass there.

Python 2.7 pass there but python 3 don't.

Example:

```
$ tox -e py38
...
pymemcache/test/test_utils.py::test_prepand_append PASSED

----------- coverage: platform linux, python 3.8.0-final-0 -----------
Name                              Stmts   Miss  Cover   Missing
---------------------------------------------------------------
pymemcache/__init__.py               12      0   100%
pymemcache/client/__init__.py        10      0   100%
pymemcache/client/base.py           614     73    88%   104, 170, 362, 364, 397-403, 408, 634, 669, 688, 717, 773, 777,
 919, 989-990, 995-997, 1028-1030, 1212, 1214, 1227, 1248, 1280-1281, 1290, 1298-1302, 1307-1314, 1317-1324, 1354-1361,
 1364-1365, 1372-1376, 1379-1380, 1383, 1388, 1392, 1440, 1471, 1495
pymemcache/client/hash.py           240     34    86%   207-212, 227, 234, 243-266, 279-286, 293, 320-323, 425, 436, 43
9, 452, 455, 461-462
pymemcache/client/murmur3.py         33      0   100%
pymemcache/client/rendezvous.py      25      0   100%
pymemcache/exceptions.py             14      0   100%
pymemcache/fallback.py               58     58     0%   15-123
pymemcache/pool.py                   88     24    73%   37, 41, 48, 52, 62, 83, 98-100, 110-112, 115-127
pymemcache/serde.py                  75      9    88%   61-62, 93-96, 103-107
---------------------------------------------------------------
TOTAL                              1169    198    83%
Coverage XML written to file coverage.xml


========================================= 403 passed, 116 deselected in 2.90s =========================================
_______________________________________________________ summary _______________________________________________________
  py38: commands succeeded
  congratulations :)
```

As this is related to unicode/bytes I suppose that some part here are related to python 2 specificities.

Reply:
done

Reply:
Thanks